azure/scope/cluster_test.go

```diff
@@ -63,6 +65,17 @@ func specArrayToString(specs []azure.ResourceSpecGetter) string {
 	return sb.String()
 }
 
+func specArrayToStringForAsoResource[T genruntime.MetaObject](specs []azure.ASOResourceSpecGetter[T]) string {
```

Comment:
Likewise, I don't think we need this function if we modify the existing `specArrayToString` to look like:
```suggestion
func specArrayToStringForAsoResource[T any](specs []T) string {
```

Reply:
Updated and force pushed.